python_modules/libraries/dagster-gcp/dagster_gcp/bigquery/io_manager.py

```diff
@@ -0,0 +1,226 @@
+
+        If you cannot authenticate with GCP via a standard method
+        (see https://cloud.google.com/docs/authentication/provide-credentials-adc), you can provide a
+        service account key as the "gcp_credentials" configuration. Dagster will store this key in a
+        temporary file and set GOOGLE_APPLICATION_CREDENTIALS to point to the file. After the run completes,
+        the file will be deleted, and GOOGLE_APPLICATION_CREDENTIALS will be unset.
+
+    """
+
+    @io_manager(
+        config_schema={
+            "dataset": Field(
+                StringSource,
+                description=(
+                    "Name of the BigQuery dataset to use. If not provided, the last prefix before"
+                    " the asset name will be used."
+                ),
+                is_required=False,
+            ),
+            "project": Field(StringSource, description="The GCP project to use."),
+            "location": Field(
+                Noneable(StringSource),
+                is_required=False,
+                default_value=None,
+                description="The GCP location.",
+            ),
+            "gcp_credentials": Field(
+                Noneable(StringSource),
+                is_required=False,
+                default_value=None,
+                description=(
+                    "GCP authentication credentials. If provided, a temporary file will be created"
+                    " with the credentials and GOOGLE_APPLICATION_CREDENTIALS will be set to the"
+                    " temporary file."
+                ),
+            ),
+        }
+    )
+    def bigquery_io_manager(init_context):
+        """I/O Manager for storing outputs in a BigQuery database
+
+        Assets will be stored in the dataset and table name specified by their AssetKey.
+        Subsequent materializations of an asset will overwrite previous materializations of that asset.
+        Op outputs will be stored in the dataset specified by output metadata (defaults to public) in a
+        table of the name of the output.
+
+        Note that the BigQuery config is mapped to the DB IO manager table hierarchy as follows:
+        * project -> database
+        * dataset -> schema
+        * table -> table
+        """
+        temp_file_name = ""
+        if init_context.resource_config.get("gcp_credentials"):
+            with tempfile.NamedTemporaryFile("w", delete=False) as f:
+                f.write(init_context.resource_config.get("gcp_credentials"))
+                temp_file_name = f.name
+                os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = temp_file_name
+
+        yield DbIOManager(
+            type_handlers=type_handlers,
+            db_client=BigQueryClient(),
+            io_manager_name="BigQueryIOManager",
+            database=init_context.resource_config["project"],
+            schema=init_context.resource_config.get("dataset"),
+        )
+        if init_context.resource_config.get("gcp_credentials"):
```

Comment:
I think this should be in a finally block, right?

Reply:
yeah i tried that at first and remember getting an error about not being a context manager, but i tried it again and now it works! so it must have been actually caused by something else. I'll make it a finally, thanks for making me try it again